tools/config.py

```diff
@@ -254,20 +248,27 @@ def generate_config(path, first_time=False):
 elif os.path.exists(user_home_config):
   EM_CONFIG = user_home_config
 else:
+  # No config file found.  Set EM_CONFIG to a default value
+  # that will get reported in the error below.
   if root_is_writable():
-    generate_config(embedded_config, first_time=True)
+    EM_CONFIG = embedded_config
   else:
-    generate_config(user_home_config, first_time=True)
-  sys.exit(0)
+    EM_CONFIG = user_home_config
 
 # We used to support inline EM_CONFIG.
 if '\n' in EM_CONFIG:
   exit_with_error('Inline EM_CONFIG data no longer supported.  Please use a config file.')
 
 EM_CONFIG = os.path.expanduser(EM_CONFIG)
-logger.debug('emscripten config is located in ' + EM_CONFIG)
+
+if '--generate-config' in sys.argv:
+  generate_config(EM_CONFIG)
+  sys.exit(0)
```

Comment:
why move this from emcc.py? so other scripts can use it?

Reply:
Because importing `config.py` will cause the error regarding the missing config file to bring down the application before its gets to the argv parsing stage.  

This is the issue in #13962.. you can't call `--generate-config` unless you already have a config with the current code :(